kafka-rest/src/test/java/io/confluent/kafkarest/v2/LoadTest.java

```diff
@@ -0,0 +1,172 @@
+package io.confluent.kafkarest.v2;
+
+import io.confluent.kafkarest.ConsumerReadCallback;
+import io.confluent.kafkarest.KafkaRestConfig;
+import io.confluent.kafkarest.SystemTime;
+import io.confluent.kafkarest.entities.BinaryConsumerRecord;
+import io.confluent.kafkarest.entities.ConsumerInstanceConfig;
+import io.confluent.kafkarest.entities.ConsumerRecord;
+import io.confluent.kafkarest.entities.ConsumerSubscriptionRecord;
+import io.confluent.kafkarest.entities.EmbeddedFormat;
+import io.confluent.rest.exceptions.RestException;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.OffsetResetStrategy;
+import org.apache.kafka.common.TopicPartition;
+import org.easymock.Capture;
+import org.easymock.EasyMock;
+import org.easymock.EasyMockRunner;
+import org.easymock.IExpectationSetters;
+import org.easymock.Mock;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Properties;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
+
+@RunWith(EasyMockRunner.class)
+public class LoadTest {
+
+    private KafkaRestConfig config;
+    @Mock
+    private KafkaConsumerManager.KafkaConsumerFactory consumerFactory;
+
+    private KafkaConsumerManager consumerManager;
+
+    private static final String topicName = "testtopic";
+
+    private Capture<Properties> capturedConsumerConfig;
+
+    private long requestTimeoutMs = 1000;
+
+    class ConsumerTestRun {
+        private final MockConsumer consumer;
+        private volatile boolean sawCallback;
+        private ConsumerReadCallback callback;
+        private RestException actualException;
```

Comment:
I guess this and `actualRecords` should be volatile also?

Reply:
Yeah. I don't think volatile should have much effect here since we're doing things in a serial manner - maybe for the memory barrier it's useful. I added the original in good taste